### PR TITLE
Ci tests (#2)

### PR DESCRIPTION
**Pull request by emilpriver**
hey 👏

Added Ci to build the project on: 
- Ubuntu 22.04
- Ubuntu 20.04
- macos 12
- macos 11
- Windows 2022 (server)
- Windows 2019 (server)

Don't seem as we are able to use for example windows 10 but windows server does the job aswell.

I also added a linter which complains about normal problems: https://github.com/rust-lang/rust-clippy
and `rust fmt` check.

### CODE REVIEW

**Comment by emilpriver** [comment]
It's is possible to build/test into different rust targets. But don't think it needed atm

**Comment by emilpriver** [comment]
Did see that GH is not running the actions unless you merge it @benja but here is a example of all tests: https://github.com/emilpriver/cloup/actions/runs/3269186840

**Comment by emilpriver** [comment]
I recommend to Squash this PR

File: .github/workflows/test-rust.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+          rustup install stable          
+      - name: Toolchain info
+        run: |
+          cargo --version --verbose
+          rustc --version
+      - name: Fetch cargo deps
+        run: cargo fetch
+      - name: Run cargo build
+        run: cargo build
+      - name: Cache deps
```
Comment (Brendonovich) [inline .github/workflows/test-rust.yaml]: Might be better to use https://github.com/Swatinem/rust-cache

File: src/commands/init.rs [hunk 2]
```diff
@@ -10,11 +10,8 @@ pub fn run(_namespace: Option<String>) {
         .join("cloup");
 
     if let Err(e) = fs::create_dir(&config_dirname) {
-        match e.kind() {
-            ErrorKind::PermissionDenied => {
-                eprintln!("Permission denied when creating config directory")
-            }
-            _ => (),
+        if e.kind() == ErrorKind::PermissionDenied {
```
Comment (Brendonovich) [inline src/commands/init.rs]: Probably worth keeping the match statement for if other error types want to be handled in the future.

**Review by Brendonovich — CHANGES_REQUESTED** [review]
Wanted to get these review notes down before Benja forgets them

File: src/commands/init.rs (hunk 2 above)
Comment (emilpriver) [reply src/commands/init.rs]: Got it. I've re-added this for upcoming commit

File: .github/workflows/test-rust.yaml (hunk 1 above)
Comment (emilpriver) [reply .github/workflows/test-rust.yaml]: Didn't know about this actions so replaced it!

File: src/commands/init.rs (hunk 2 above)
Comment (emilpriver) [reply src/commands/init.rs]: At the same time do I think that it’s good to rewrite the code to use match later when is needed and use if statement now

**Pull request closed without merging**